src/plots/xyPlot.ts

```diff
@@ -109,7 +111,7 @@ export class XYPlot<X, Y> extends Plot {
       return this._deferredRendering;
     }
 
-    if (deferredRendering) {
+    if (deferredRendering && this._isAnchored) {
```

Comment:
This is the main fix. There is no point in cacheing the domain when the plot is not anchored, because the domain of the scale will be something default (like [0, 1])